src/server/services/wmts/qgswmtsparameters.h

```diff
@@ -0,0 +1,328 @@
+/***************************************************************************
+                              qgswmtsparameters.h
+                              -------------------
+  begin                : Aug 10, 2018
+  copyright            : (C) 2018 by René-Luc Dhont
+  email                : rldhont at 3liz dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSWMTSPARAMETERS_H
+#define QGSWMTSPARAMETERS_H
+
+#include <QMap>
+#include <QObject>
+#include <QMetaEnum>
+
+#include "qgswmtsserviceexception.h"
+#include "qgsserverrequest.h"
+#include "qgsprojectversion.h"
+#include "qgsserverparameters.h"
+
+namespace QgsWmts
+{
+
+  /**
+   * \ingroup server
+   * \class QgsWmts::QgsWmsParameter
+   * \brief WMS parameter used by WMTS service.
+   * \since QGIS 3.4
+   */
+  class QgsWmsParameter : public QgsServerParameterDefinition
```

Comment:
I'm really not convinced by redefining a `QgsWmsParameter` here. Why can we not use the class defined in wms service instead?

Reply:
I failed to link WMS module to WMTS module. And I prefer to defined a restricted list of WMS parameter.

Reply:
> And I prefer to defined a restricted list of WMS parameter.

OK. Can we have another name then? Like `QgsWmsParameterForWmts` (or anything like that)? Just to avoid confusion.